apps/api/src/app/events/usecases/trigger-event/message-filter.matcher.ts

```diff
@@ -145,9 +165,44 @@ async function getWebhookResponse(i, variables: IFilterVariables): Promise<Recor
   }
 }
 
-async function processFilter(variables: IFilterVariables, i) {
+async function buildPayload(variables: IFilterVariables, configuration: IMessageFilterConfiguration) {
+  const payload: Partial<{
+    subscriber: SubscriberEntity;
+    payload: Record<string, unknown>;
+    hmac: string;
+    identifier: string;
```

Comment:
Let's also add `channel` here and `providerId` optionally?

Reply:
https://github.com/novuhq/novu/pull/2348/commits/da056a25e1ec91e2557f27199a4b16ae3cfa6923